### PR TITLE
Reintroduce engine get client version v1 with commit in manifest

### LINKED ISSUE #6471: engine_getClientVersionV1 - expose web3_clientVersion via EngineAPI
### Description
As an Consensus Client, I want access to my execution clients name and version over the Engine API so that I can include it in my proposed block grafitti.

https://github.com/ethereum/execution-apis/blob/main/src/engine/identification.md

### Acceptance Criteria
* Proposed blocks from any CL include the besu version by default.

### PR DESCRIPTION
**Pull request by Matilda-Clerke**
re-fixes #6471 

for reference:
https://github.com/ethereum/execution-apis/blob/main/src/engine/identification.md

### CODE REVIEW

**Comment by Matilda-Clerke** [comment]
Note: CHANGELOG.md may need to be updated if this is merged after the next release

File: config/build.gradle [hunk 1]
```diff
@@ -22,7 +22,8 @@ jar {
       'Specification-Title': archiveBaseName,
       'Specification-Version': project.version,
       'Implementation-Title': archiveBaseName,
-      'Implementation-Version': calculateVersion()
+      'Implementation-Version': calculateVersion(),
+      'Commit-Hash': getGitCommitDetails(40).hash
```
Comment (garyschulte) [inline config/build.gradle:26]: are we doing this everywhere as a matter of consistency?  It seems like we only currently need this in the `:besu` artifact, but I don't see a reason not to be consistent across all

File: build.gradle [hunk 2]
```diff
@@ -823,7 +823,7 @@ task distDocker {
         dockerPlatform = "--platform ${project.getProperty('docker-platform')}"
         println "Building for platform ${project.getProperty('docker-platform')}"
       }
-      def gitDetails = getGitCommitDetails(7)
+      def gitDetails = getGitCommitDetails()
```
Comment (garyschulte) [inline build.gradle:826]: 8 seems fine, but if we are going to generate a sha that is different from the build artifact name, we may as well use all 40 here

**Review by garyschulte — APPROVED** [review]
LGTM, nothing blocking (except pesky failing info regex tests)

File: config/build.gradle (hunk 1 above)
Comment (Matilda-Clerke) [reply config/build.gradle:26]: Yes, I updated all of the gradle sub-modules where we're setting Implementation-Version to also set a Commit-Hash for consistency. Currently, it's only used in the besu module

**Pull request merged**